### PR TITLE
docs: consistent case (sentence)

### PR DESCRIPTION
**Pull request by froystig**
More consistent, and easier to read, especially around code tokens and technical terms.

### CODE REVIEW

**Review by jakevdp — APPROVED** [review]
Looks good!

One thing I confirmed: auto-generated link anchors are already normalized to lowercase on generated pages, so this shouldn't break any links (e.g. https://jax.readthedocs.io/en/latest/notebooks/thinking_in_jax.html#static-vs-traced-operations)
